sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -472,10 +423,13 @@ export class MessageSender extends LinkEntity {
    */
   async onDetached(senderError?: AmqpError | Error): Promise<void> {
     try {
-      const wasCloseInitiated = this._sender && this._sender.isItselfClosed();
+      const wasCloseInitiated = this.link && this.link.isItselfClosed();
```

Comment:
We should be able to use `this.wasClosed` here.

When we refactored the `onDetached()` code for the streaming receiver to use a local flag than rely on `isItselfClosed()`, we left the sender untouched. There is no reason why the `onDetached()` method for the sender should be different from that of the streaming receiver other than the fact that there is no reporting of the error to the user here

Reply:
Done.